### DISCUSSION TITLE
Is `auth` option secure?

### DISCUSSION THREAD

**Discussion by ghost** [socketio/socket.io#4627] [Q&A] [answered]
Hi,
I was wondering how the `auth` option is handled by socket.io in different use cases.
Particularly:
- When using `long-polling` as transport method where are the credentials encoded? Are they encoded on some HTTP request's body?
- The same question in the `websocket` transport method

I'm trying to understand if this option is secure over a HTTPS connection, so the credentials are not encoded neither in a querystring nor other unsafe places but they are encrypted over a secure connection.

**Comment by darrachequesne** [comment, marked as answer]
Hi!

The `auth` option is sent during the handshake, in a CONNECT packet:

- with HTTP long-polling, the credentials are indeed sent in the request body of a POST request
- with WebSocket, the credentials are sent in a WebSocket frame

Format: `<Engine.IO MESSAGE packet type><Socket.IO CONNECT packet type><JSON.stringified auth>`
Example: `40{"token":"abc"}`

Reference: https://github.com/socketio/socket.io-protocol

**Reply by ghost** [reply]
Thank you @darrachequesne, so in both cases the credentials gets encrypted over a SSL/TLS connection?

**Reply by darrachequesne** [reply]
Yes, that's right.

**Reply by ghost** [reply]
Thank you very much!
